ts/src/header-validator/validate-json.ts

```diff
@@ -725,6 +726,101 @@ const priorityField: StructFields<Priority> = {
   priority: field('priority', int64, 0n),
 }
 
+function isDebugTypeSupported(ctx: RegistrationContext, s: string) {
+  return ctx.aggregatableDebugTypes.includes(s)
+}
+
+function aggregatableDebugType(
+  ctx: RegistrationContext,
+  j: Json
+): Maybe<string> {
+  return string(ctx, j).peek((s) => {
+    if (!isDebugTypeSupported(ctx, s)) {
+      ctx.warning('unknown type')
+    }
+  })
+}
+
+export type KeyPiece = {
+  keyPiece: bigint
+}
+
+const keyPieceField: StructFields<KeyPiece> = {
+  keyPiece: field('key_piece', hex128),
+}
+
+export type AggregatableDebugReportingData = KeyPiece & {
+  types: string[]
```

Comment:
Should we make this a set instead of an array?

Reply:
We treat supported and unsupported types differently. Unsupported types are just ignored, no matter whether it duplicates or not, but we error for duplicated supported types. Therefore, it would be easier to use an array and validate later.

Reply:
OK, that's fine for now, but I may try to clean this up later.